### PR TITLE
Rework Steam detection (part 1)

### PR DESCRIPTION
**Pull request by 0xA50C1A1**
Please sign (check) the below before submitting the Pull Request:

- [x] I have signed the ntop Contributor License Agreement at https://github.com/ntop/legal/blob/main/individual-contributor-licence-agreement.md
- [x] I have read the contributing guide lines at https://github.com/ntop/nDPI/blob/dev/CONTRIBUTING.md
- [x] I have updated the documentation (in doc/) to reflect the changes made (if applicable)

Description:

Steam dissector code has been bloated by signature checks of the protocol used by [ancient](https://github.com/koraktor/steam-condenser/issues/212) [versions of Source Engine](https://developer.valvesoftware.com/wiki/Talk:Server_queries). I removed them because Steam now only uses TLS connections, and all modern Valve games (i.e. the ones that still get updates) use the Steam Datagram Relay protocol.

I also removed the CS:GO dissector, since it was broken by the last protocol changes and now it's just not needed.

A detailed discussion of the problem can be found at #2262.

Plans for part 2: implement Steam Remote Play detection and improve Steam Datagram Relay detection.

### CODE REVIEW

File: src/include/ndpi_typedefs.h [hunk 1]
```diff
@@ -1437,9 +1437,6 @@ struct ndpi_flow_struct {
   /* NDPI_PROTOCOL_RTMP */
   u_int8_t rtmp_stage:2;
 
-  /* NDPI_PROTOCOL_STEAM */
-  u_int16_t steam_stage:3, steam_stage1:3, steam_stage2:2, steam_stage3:2;
-
```
Comment (IvanNardi) [inline src/include/ndpi_typedefs.h:1446]: Could you also remove the similar csgo fields?

File: src/lib/protocols/steam.c [hunk 2]
```diff
@@ -1,11 +1,7 @@
 /*
  * steam.c
  *
- * Copyright (C) 2011-22 - ntop.org
- * Copyright (C) 2014 Tomasz Bujlow <tomasz@skatnet.dk>
```
Comment (IvanNardi) [inline src/lib/protocols/steam.c:5]: Could you keep these 2 lines, please (updating 22->24)?

File: src/lib/protocols/steam_datagram_relay.c [hunk 3]
```diff
@@ -0,0 +1,71 @@
+/*
+ * steam_datagram_relay.c
+ *
+ * Copyright (C) 2023 - ntop.org
+ * Copyright (C) 2023 - V.G <jacendi@protonmail.com>
```
Comment (IvanNardi) [inline src/lib/protocols/steam_datagram_relay.c]: 2024 in both lines :)

**Comment by 0xA50C1A1** [comment]
@IvanNardi Thanks for the review. As usual, I missed some things :)

**Review by IvanNardi — APPROVED** [review]

**Comment by IvanNardi** [comment]
@0xA50C1A1, could you rebase, please? You should update some results of some new tests...

**Comment by 0xA50C1A1** [comment]
> @0xA50C1A1, could you rebase, please? You should update some results of some new tests...

Done.

File: tests/cfgs/default/result/ossfuzz_seed_fake_traces_1.pcapng.out [hunk 4]
```diff
@@ -22,24 +23,28 @@ Patricia risk IPv6:   0/0 (search/found)
...
+	9	TCP 192.168.1.128:1 -> 202.9.66.76:1119 [proto: 213/Starcraft][IP: 213/Starcraft][ClearText][Confidence: DPI][DPI packets: 1][cat: Game/8][1 pkts/50 bytes -> 0 pkts/0 bytes][Goodput ratio: 20/0][< 1 sec][Risk: ** Unidirectional Traffic **** TCP Connection Issues **][Risk Score: 60][Risk Info: TCP NULL scan / No client to server traffic][Plen Bins: 100,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
+
+
+Undetected flows:
+	1	UDP 127.0.0.1:100 <-> 127.0.0.1:200 [proto: 0/Unknown][IP: 0/Unknown][ClearText][Confidence: Unknown][DPI packets: 2][1 pkts/32 bytes <-> 1 pkts/36 bytes][Goodput ratio: 12/22][45.10 sec][Plen Bins: 100,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
```
Comment (utoni) [inline tests/cfgs/default/result/ossfuzz_seed_fake_traces_1.pcapng.out:50]: Aren't we loosing Steam detection here? Or is that PCAP too old to care about?
// EDIT: Just saw it's a fuzzing trace...

File: src/lib/protocols/steam.c [hunk 5]
```diff
@@ -24,303 +21,46 @@
...
-    }
-		
-    if (ndpi_match_strprefix(packet->payload, payload_len, "\xff\xff\xff\xff")) {
-      NDPI_LOG_DBG2(ndpi_struct, "Possible STEAM request detected, we will look further for the response..\n");
-
-      /* Encode the direction of the packet in the stage, so we will know when we need to look for the response packet. */
-      flow->steam_stage1 = packet->packet_direction + 3; // packet_direction 0: stage 3, packet_direction 1: stage 4
-      return;
-    }
+  /* TODO: implement Steam Remote Play detection */
```
Comment (utoni) [inline src/lib/protocols/steam.c:51]: Do you have any plans to implement that one?
// EDIT: Part2? :D

Comment (0xA50C1A1) [reply src/lib/protocols/steam.c:51]: > Do you have any plans to implement that one? // EDIT: Part2? :D

Yeah, I plan to do it when I get some spare time. I found [some](https://blog.thalium.re/posts/achieving-remote-code-execution-in-steam-remote-play/) [descriptions](https://github.com/mariotaku/steamlink.py/tree/master/session) (unofficial ofс) of the protocol, but anyway I'll have to start streaming to capture the traffic.

**Review by utoni — APPROVED** [review]

**Comment by IvanNardi** [comment]
Thanks

**Pull request merged**